eth-contracts/contracts/TokenLocker.sol

```diff
@@ -1,95 +1,99 @@
 pragma solidity ^0.5.10;
+pragma experimental ABIEncoderV2;
 
 import "./interfaces/IERC20.sol";
+import {CKBCrypto} from "./libraries/CKBCrypto.sol";
 import {TypedMemView} from "./libraries/TypedMemView.sol";
 import {SafeMath} from "./libraries/SafeMath.sol";
-import {ViewCKB} from "./libraries/ViewCKB.sol";
+import {CKBTxView} from "./libraries/CKBTxView.sol";
 import {ViewSpv} from "./libraries/ViewSpv.sol";
 import {Address} from "./libraries/Address.sol";
 import {ICKBSpv} from "./interfaces/ICKBSpv.sol";
 
 contract TokenLocker {
     using SafeMath for uint256;
     using Address for address;
-
     using TypedMemView for bytes;
     using TypedMemView for bytes29;
-    using ViewCKB for bytes29;
+    using CKBTxView for bytes29;
     using ViewSpv for bytes29;
 
+    struct BurnResult {
+        uint256 amount;
+        address token;
+        address recipient;
+    }
+
     uint64 public numConfirmations_;
     ICKBSpv public ckbSpv_;
+    bytes32 public recipientCellTypescriptCodeHash_;
+    uint8 public recipientCellTypescriptHashType_;
 
     // txHash -> Used
     mapping(bytes32 => bool) public usedTx_;
 
     event Locked(
         address indexed token,
         address indexed sender,
+        string  indexed ckbAddress,
```

Comment:
ckbAddress should not be ```indexed```, if a ```string``` type is indexed, it's hash will be stored in topics, thus proof generator can not decode string from topic

Reply:
right